# HG changeset patch
# User Rik <rik@octave.org>
# Date 1382448774 25200
#      Tue Oct 22 06:32:54 2013 -0700
# Node ID 132d9f543cc5175ec099f05d48123ffe624b6e2f
# Parent  5b0036d89d4adcc6b37708b49e46c2f10360d72f
print.m: 2nd fix for printing when using -F option (bug #40334)

* scripts/plot/util/print.m: Apply fontsize attribute to non-axes objects
first to avoid triggering legend fontsize callback which destroys objects
in the list.

diff --git a/scripts/plot/util/print.m b/scripts/plot/util/print.m
--- a/scripts/plot/util/print.m
+++ b/scripts/plot/util/print.m
@@ -400,18 +400,22 @@ function print (varargin)
           fontsize = str2double (opts.fontsize);
         else
           fontsize = opts.fontsize;
         endif
         if (! isempty (opts.scalefontsize) && ! opts.scalefontsize != 1)
           ## This is done to work around the bbox being whole numbers.
           fontsize = fontsize * opts.scalefontsize;
         endif
-        ## FIXME: sort needed so legend child objects are acted on first
-        set (sort (h(ishandle (h))), "fontsize", fontsize);
+        ## FIXME: legend child objects need to be acted on first.
+        ##        or legend fontsize callback will destroy them.
+        hlist = h(ishandle (h));
+        haxes = strcmp (get (hlist, "type"), "axes");
+        set (hlist(! haxes), "fontsize", fontsize);
+        set (hlist(haxes), "fontsize", fontsize);
       endif
     endif
 
     ## call the graphcis toolkit print script
     switch (get (opts.figure, "__graphics_toolkit__"))
       case "gnuplot"
         opts = __gnuplot_print__ (opts);
       otherwise
